async_service/base.py

```diff
@@ -42,7 +62,7 @@ def __init__(self, service: ServiceAPI) -> None:
         if hasattr(service, "manager"):
```

Comment:
Isn't this always going to return True now that `manager` is an @property?

Reply:
Oh, I guess it will return False because the `manager` `@property` raises an AttributeError when `_manager` is not set. Maybe I just need to get used to that, but it wasn't immediately obvious to me.

Reply:
I mean it was surprising because `hasattr` actually accesses the `manager` attribute instead of just checking if it exists. I don't know why I thought it did the latter

Reply:
Yeah, I think this should be checking the `_manager` property instead.